python/lsst/ts/adamSensors/adamSensorsCSC.py

```diff
@@ -26,23 +34,98 @@ def __init__(self, config_dir=None, initial_state=salobj.State.STANDBY, initial_
+            2: (self.config.analog_input_2_type, poly1d(self.config.analog_input_2_coefficients)),
+            3: (self.config.analog_input_3_type, poly1d(self.config.analog_input_3_coefficients)),
+            4: (self.config.analog_input_4_type, poly1d(self.config.analog_input_4_coefficients)),
+            5: (self.config.analog_input_5_type, poly1d(self.config.analog_input_5_coefficients)),
+        }
 
+        # figure out which topics to publish
+        hasTemperature = False
+        hasPressure = False
+        for s in sensors:
+            if sensors[s][0] == "Pressure":
+                hasPressure = True
+            if sensors[s][0] == "Temperature":
+                hasTemperature = True
+
+        loop = asyncio.get_event_loop()
+        executor = concurrent.futures.ThreadPoolExecutor()
+
+        outputs = [0, 0, 0, 0, 0, 0]
         while self.connected:
-            voltages = self.adam.read_voltage()
+            voltages = await loop.run_in_executor(executor, self.adam.read_voltage)
+
+            # convert the voltage into whatever units, according to the
+            # polynomial defined in configuration
+            for i in range(6):
+                outputs[i] = sensors[i][1](voltages[i])
+
+            # Assemble telemetry topics
+            # Channel 0
+            if sensors[0][0] == "Pressure":
+                self.tel_pressure.set(pressure_ch0=outputs[0])
+            elif sensors[0][0] == "Temperature":
+                self.tel_temperature.set(temp_ch0=outputs[0])
+
+            # Channel 1
+            if sensors[1][0] == "Pressure":
+                self.tel_pressure.set(pressure_ch1=outputs[1])
+            elif sensors[1][0] == "Temperature":
+                self.tel_temperature.set(temp_ch1=outputs[1])
+
+            # Channel 2
+            if sensors[2][0] == "Pressure":
+                self.tel_pressure.set(pressure_ch2=outputs[2])
+            elif sensors[2][0] == "Temperature":
+                self.tel_temperature.set(temp_ch2=outputs[2])
+
+            # Channel 3
+            if sensors[3][0] == "Pressure":
+                self.tel_pressure.set(pressure_ch3=outputs[3])
+            elif sensors[3][0] == "Temperature":
+                self.tel_temperature.set(temp_ch3=outputs[3])
+
+            # Channel 4
+            if sensors[4][0] == "Pressure":
+                self.tel_pressure.set(pressure_ch4=outputs[4])
+            elif sensors[4][0] == "Temperature":
+                self.tel_temperature.set(temp_ch4=outputs[4])
+
+            # Channel 5
+            if sensors[5][0] == "Pressure":
+                self.tel_pressure.set(pressure_ch5=outputs[5])
+            elif sensors[5][0] == "Temperature":
+                self.tel_temperature.set(temp_ch5=outputs[5])
+
+            # publish telemetry
+            if hasPressure:
+                self.tel_pressure.put()
+            if hasTemperature:
+                self.tel_temperature.put()
 
+            asyncio.sleep(1)
```

Comment:
Use the heartbeat time here:

```
asyncio.sleep(self.heartbeat_interval)
```

This is already part of the CSC interface so no need to define (https://github.com/lsst-ts/ts_salobj/blob/20d0acbee634e9f052567854e114314ff382d88f/python/lsst/ts/salobj/base_csc.py#L184).